docs/analyzer/recognizer_registry_provider.md

```diff
@@ -0,0 +1,92 @@
+# Customizing recognizer registry
+To load recognizers from file, use `RecognizerRegistryProvider` to instantiate the recognizer registry and then pass it through to the analyzer engine:
+
+```python
+from presidio_analyzer import AnalyzerEngine
+from presidio_analyzer.recognizer_registry import RecognizerRegistryProvider
+
+recognizer_registry_conf_file = "./analyzer/recognizers-config.yml"
+
+provider = RecognizerRegistryProvider(
+                conf_file=recognizer_registry_conf_file
+            )
+registry = provider.create_recognizer_registry()
+analyzer = AnalyzerEngine(
+            registry=registry,
+            )
+
+results = analyzer.analyze(text="My name is Morris")
+print(results)
+```
+
+## Configuration file structure
+
+```yaml
+global_regex_flags: 26
+recognizers: 
+...
+```
+
+The configuration file consists of two parts:
+
+  - `global_regex_flags`: regex flags to be used in regex matching.
+  - `recognizers`: a list of recognizers to be loaded by the recognizer registry. This list consists of two different types of recognizers: 
+    - Predefined: A set of already defined recognizer classes in presidio
+    - Custom: custom created pattern recognizers that are created ad hoc based on the fields provided in the configuration file
+
+## Recognizer list
+
+The recognizer list comprises of both the predefined and custom recognizers, for example: 
+
+```yaml
+...
+    - name: CreditCardRecognizer
+    supported_languages:
+    - language: en
+        context: [credit, card, visa, mastercard, cc,
+        amex, discover, jcb, diners, maestro, instapayment]
+    - language: es
+        context: [tarjeta, credito, visa, mastercard, cc,
+        amex, discover, jcb, diners, maestro, instapayment]
+    - language: it
+    - language: pl
+    type: predefined
+
+    - name: UsBankRecognizer
+    supported_languages: 
+    - en
+    type: predefined
+
+    - MedicalLicenseRecognizer
+
+    - name: ExampleCustomRecognizer
+    patterns:
+        - name: "zip code (weak)"
+        regex: "(\\b\\d{5}(?:\\-\\d{4})?\\b)"
+        score: 0.01
+        - name: "zip code (weak)"
+        regex: "(\\b\\d{5}(?:\\-\\d{4})?\\b)"
+        score: 0.01
+    supported_languages:
+        - language: en
+        context: [credit]
+        - language: es
+        context: [tarjeta,credito]
+    supported_entity: "ZIP"
+    deny_list: [Mr., Mrs., Ms., Miss, Dr., Prof.]
+    deny_list_score: 1
+    type: custom
+    enabled: true
+```
+
+The recognizer parameters:
+
+  - `supported_languages`: A list of supported languages that the analyzer will support. In case this field is missing, a recognizer will be created for each supported language provided be `AnalyzerEngine`. 
+  In addition to the language code, this field also contains the context, which increases confidence in the detection (as seen in the credit card example above).
```

Comment:
```suggestion
  In addition to the language code, this field also contains a list of context words, which increases confidence in the detection in case it is found in the surroundings of a detected entity (as seen in the credit card example above).
```